pkg/odo/cli/component/create.go

```diff
@@ -447,6 +447,10 @@ func (co *CreateOptions) Complete(name string, cmd *cobra.Command, args []string
 
 		// Configure the default namespace
 		var defaultComponentNamespace string
+		client, err := genericclioptions.Client()
+		if err == nil {
+			defaultComponentNamespace = client.GetCurrentProjectName()
+		}
```

Comment:
Since `defaultComponentNamespace`  is only used before `ui.EnterDevfileComponentProject()`, it might be a good idea to delay the variable and client initialization  and put it right before it.